main.js

```diff
@@ -1,6 +1,10 @@
 // API
 import "./lib/transform.js";
-export { minify } from "./lib/minify.js";
+import { minify } from "./lib/minify.js";
+export { minify };
+// Also re-export the API as default export to keep compatibility with existing CommonJS tooling.
+// See https://github.com/terser-js/terser/issues/251 for more info.
+export default { minify };
```

Comment:
For full compat maybe even exports the full namespace object instead of only the `minify` function.

``` js
export * as default from "./main";
```

Reply:
You mean in another file, right? Maybe something like:
- move contents of `main.js` to `api.js`
- current `main.ts` becomes:
```
export * from "./api";
export * as default from "./api";
```

I think that makes sense, but would like to know what @fabiosantoscode thinks of the extra file.

Reply:
Let's fix plugin instead
https://github.com/webpack-contrib/terser-webpack-plugin/blob/master/src/minify.js#L4

Reply:
> You mean in another file, right? 

I really meant this file. This reference the exports of the file itself. Yes it's circular, but fine here.

Extra file would also work.

> Let's fix plugin instead

Also possible and we probably will do this if this won't get fixed here, but keep in mind that the problem was caused by a breaking change released as minor version and terser-webpack-plugin is probably not the only one consuming terser this way. Other libs are probably also broken because of this breaking change, maybe unnoticed because they are less often used.

If the intend is to not add such a compat thing that's also fine, but it would be great to release the commit as major version instead (which may makes sense anyway as it also increases the minimum supported node.js version from 4 to 6).

Reply:
@sokra couldn't get it to work with the syntax your provided:
```
[!] Error: Unexpected token
main.js (47:9)
45:
46:
47: export * as default from "./main";
             ^
Error: Unexpected token
```

Instead used this:
```
import * as terser from './main';
export default terser;
```
 PR is updated.